presto-hive/src/main/java/com/facebook/presto/hive/NumberParser.java

```diff
@@ -41,18 +41,34 @@ public static long parseLong(byte[] bytes, int start, int length)
         return value * sign;
     }
 
-    public static double parseDouble(byte[] bytes, int start, int length)
+    public static float parseFloat(byte[] bytes, int start, int length)
     {
-        char[] chars = new char[length];
-        for (int pos = 0; pos < length; pos++) {
-            chars[pos] = (char) bytes[start + pos];
+        String string = getStringOfBytes(bytes, start, length);
+        try {
+            return Float.parseFloat(string);
         }
-        String string = new String(chars);
+        catch (NumberFormatException e) {
+            throw new PrestoException(HIVE_BAD_DATA, e);
+        }
+    }
+
+    public static double parseDouble(byte[] bytes, int start, int length)
+    {
+        String string = getStringOfBytes(bytes, start, length);
         try {
             return Double.parseDouble(string);
         }
         catch (NumberFormatException e) {
             throw new PrestoException(HIVE_BAD_DATA, e);
         }
     }
+
+    private static String getStringOfBytes(byte[] bytes, int start, int length)
+    {
+        char[] chars = new char[length];
```

Comment:
The fastest way to do this is to use the deprecated String constructor `public String(byte ascii[], int hibyte, int offset, int count)`... something like `new String(bytes, 0, start, length)`